lib/cmxl/statement.rb

```diff
@@ -59,7 +59,7 @@ def reference
     end
 
     def generation_date
-      field(20).date || field(13).date
+      field(20).date || (field(13).nil? ? nil : field(13).date)
```

Comment:
I was wondering of we should do that or `field(20).date`, too? 
e.g. something like:
```ruby
field = field(20) || field(13)
field.date unless field.nil?
```

Reply:
I think it is not necessary because :20 is mandatory in MT94X. If it is not provided the complete statment is invalid and should fail before it try to parse it. If :20 is there without a date method date gives back nil. I think it should work like it is.

Reply:
ah true. yep. 
I still find the the other option more readable, but that's fine then I guess.
thanks

Reply:
I checked yesterday your proposal making it more readable.

```ruby
date_field = field(20) || field(13)
date_field.date unless field.nil?
```
Unfortunately it breaks the function. Like mentioned above :20 is mandatory so `field(20) || field(13)` will never read from :13.

Reply:
thanks for checking! 
yeah, after your comment above I am happy with that change. 

and thanks for the tests! I think that is most important, that we add test cases for everything that helps us to be sure not to break things. :)

Reply:
could you add a short comment on the code? 
something like: "get data from field 20. If the date from field 20 is nil we fall back to field 13 if present"